chunk.go

```diff
@@ -0,0 +1,43 @@
+package typhon
+
+import (
+	"io"
+	"net/http"
+)
+
+func copyChunked(dst io.Writer, src io.Reader) (written int64, err error) {
+	flusher, flusherOk := dst.(http.Flusher)
+	if !flusherOk {
+		return io.Copy(dst, src)
+	}
+
+	// This is taken and lightly adapted from the source of io.Copy
+	buf := make([]byte, 32*1024)
+	for {
+		nr, er := src.Read(buf)
+		if nr > 0 {
+			nw, ew := dst.Write(buf[0:nr])
+			if nw > 0 {
+				written += int64(nw)
+			}
+			if ew != nil {
+				err = ew
+				break
+			}
+			if flusherOk {
```

Comment:
Would this always be `true`? We check `!flusherOK` above and return, and I can't see a way this can change?

Reply:
I added this before I added the `!flusherOK` check, so you're right this is redundant now. Will change 👍